src/Microsoft.ML.Data/DataLoadSave/Binary/BinaryLoader.cs

```diff
@@ -1366,7 +1366,10 @@ public Cursor(BinaryLoader parent, Func<int, bool> predicate, Random rand)
             protected override void Dispose(bool disposing)
             {
                 if (_disposed)
```

Comment:
>_disposed [](start = 20, length = 9)

Hi @yaeldekel, thanks for looking at this...

This strikes me as odd. Certainly it seems a violation of the classic dispose pattern.

This only gets called if `_disposed` is true. I would have expected that in normal instances of the pattern, if dispose has already has been called, that it will immediately exit. To do otherwise suggests that there is *different* behavior when you call the dispose method multiple times, which is a clear sign something is seriously wrong!

Also we're calling the base dispose method at the *start* of the method call, but this is incorrect -- just as constructors are called inside out, the corresponding disposers should be called outside in -- that is, the base method should be the *last* thing called (which is, indeed, what happens).

I imagine that there's some actual bug that's happening at some level, but whatever is going on, I find it hard to believe that this is the solution?